README.md

```diff
@@ -1,9 +1,77 @@
 # asl-help
 Welcome!
 
+## Purpose
+asl-help was created to aid auto splitter authors in writing scripts for games that use the Mono framework, specifically the Unity fork. It is packaged as a .NET Framework library that can be loaded at runtime from an auto splitter script, and uses code generation to allow for the creation of more streamlined, human-readable, and update-resilient pointer paths.
+
+## Usage 
+### Development Setup
+Download the `asl-help` file located in the `lib` directory of this repository and place it in LiveSplit's components folder. This is only necessary for development — the file will be downloaded automatically for end users, provided that you configure your auto splitter correctly (see [Publishing](#publishing)).
+
+### Loading Asl-Help
+Use the following snippet to load the asl-help binary in your auto splitter. This and the following examples will assume that the game runs on the Unity engine.
+```cs
+startup {
+    Assembly.Load(File.ReadAllBytes("Components/asl-help")).CreateInstance("Unity");
+}
+```
+
+### TryLoad
+All pointers must be declared inside of a function that is assigned to the `TryLoad` member of the auto-generated `Helper` variable. This should be done inside the `init` block. If this function returns `true`, then asl-help has been successfully initialized.
+```cs
+init {
+    vars.Helper.TryLoad = (Func<dynamic, bool>)(mono => {
+        // Declare your pointers here.
+        return true;
+    });
+}
+```
+Because the `mono` variable is scoped to the anonymous function, it cannot be accessed outside of the function. Do not store a reference to it that could outlive said function.
+
+### Pointers
+Pointers in asl-help are declared with the following syntax inside of the TryLoad function.
+```cs
+vars.Helper["WatcherName"] = mono["ClassName"].Make<Type>("FieldName", offset1, offset...);
```

Comment:
This way of creating pointers is discouraged. The user should be using the `MonoHelper` overloads for making pointers. If their game is IL2CPP'd, they should be encouraged to store the class in a local.

```cs
mono.Make<T>("ClassName", "StaticFieldName", "InstanceField", "NextField", ...);
```
```cs
var c = mono["ClassName"]; // mono["Assembly", "ClassName"];

mono.Make<T>(c, "StaticFieldName", "InstanceField", "NextField", ...);

// on IL2CPP
var d = mono["DlassName"];
c.Make<T>("StaticFieldName", "InstanceField", d["NextField"], ...);
```

Reply:
Why does storing it in a local matter for il2cpp? I believe you that it does but I am just curious as to why that would make a difference.

Reply:
Also how do you specify inheritance without using the syntax that I put down?

Reply:
> Why does storing it in a local matter for il2cpp?

So that you don't attempt to query the classes every time you need them (it will be cached, but you still have to check the cache unnecessarily when you already queried it once), and to reduce on code length and increase readability.

You *can* write
```cs
var c = mono["CClass"];
var d = mono["DClass"];
var e = mono["EClass"];
var f = mono["FClass"];

vars.Helper["Value"] = c.Make<T>("StaticField", "InstanceField", d["dField"], e["eField"], f["fField]);
```
as
```cs
vars.Helper["Value"] = mono["CClass"].Make<T>("StaticField", "InstanceField", mono["DClass"]["dField"], mono["EClass"]["eField"], mono["FClass"]["fField]);
```
but that just becomes unwieldy.

Reply:
> Also how do you specify inheritance without using the syntax that I put down?

```cs
vars.Helper["Value"] = mono.Make<T>("ClassName", 1, "StaticField", "InstanceField", "NextField", ...);
```
or
```cs
var c = mono["CClass", 1];

vars.Helper["Value"] = mono.Make<T>(c, "StaticField", "InstanceField", "NextField", ...);
```